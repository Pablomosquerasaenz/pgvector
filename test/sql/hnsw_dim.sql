SET enable_seqscan = off;

CREATE TABLE t (val vector);
INSERT INTO t (val) VALUES ('[0,0,0]'), ('[1,2,3]'), ('[1,1,1]'), (NULL);
CREATE INDEX t_val_1 ON t USING hnsw (val vector_cosine_ops) WITH (dims = 3);

INSERT INTO t (val) VALUES ('[1,2,4]');
INSERT INTO t (val) VALUES ('[1,2,4,5]');

SELECT * FROM t ORDER BY val <=> '[3,3,3]';

INSERT INTO t (val) VALUES ('[1,2,4,5]');
CREATE INDEX t_val_2 ON t USING hnsw(val vector_cosine_ops) WITH (dims = 4);

TRUNCATE t;
CREATE INDEX t_val_3 ON t USING hnsw(val vector_cosine_ops) WITH (dims = 4);
INSERT INTO t (val) VALUES ('[1,2,4]');

DROP TABLE t;
