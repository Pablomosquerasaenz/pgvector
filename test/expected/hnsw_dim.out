SET enable_seqscan = off;
CREATE TABLE t (val vector);
INSERT INTO t (val) VALUES ('[0,0,0]'), ('[1,2,3]'), ('[1,1,1]'), (NULL);
CREATE INDEX t_val_1 ON t USING hnsw (val vector_cosine_ops) WITH (dims = 3);
INSERT INTO t (val) VALUES ('[1,2,4]');
INSERT INTO t (val) VALUES ('[1,2,4,5]');
ERROR:  expected 3 dimensions, not 4
SELECT * FROM t ORDER BY val <=> '[3,3,3]';
   val   
---------
 [1,1,1]
 [1,2,3]
 [1,2,4]
(3 rows)

INSERT INTO t (val) VALUES ('[1,2,4,5]');
ERROR:  expected 3 dimensions, not 4
CREATE INDEX t_val_2 ON t USING hnsw(val vector_cosine_ops) WITH (dims = 4);
ERROR:  expected 4 dimensions, not 3
TRUNCATE t;
CREATE INDEX t_val_3 ON t USING hnsw(val vector_cosine_ops) WITH (dims = 4);
INSERT INTO t (val) VALUES ('[1,2,4]');
ERROR:  expected 4 dimensions, not 3
DROP TABLE t;
